refactoring-previous-ticket-to-polymorphism

ap2144839 opened the pull request:
- method deleteElement(List<Integer> list, int i) was created in class com.peterson.collection.WorkWithCollection for deleting elements from ArrayList & LinkedList by the index;
- method deleteElement(HashSet<Integer> hashSet, Object o) was created in class com.peterson.collection.WorkWithCollection for deleting objects from HashSet;
- class com.peterson.util.Util was changed in order to use the above methods
https://trello.com/c/w8bRuNiH/8-refactoring-previous-ticket-to-polymorphism

ap2144839 committed:
refactoring-previous-ticket-to-polymorphism
- method deleteElement(List<Integer> list, int i) was created in class com.peterson.collection.WorkWithCollection for deleting elements from ArrayList & LinkedList by the index;
- method deleteElement(HashSet<Integer> hashSet, Object o) was created in class com.peterson.collection.WorkWithCollection for deleting objects from HashSet;
- class com.peterson.util.Util was changed in order to use the above methods
https://trello.com/c/w8bRuNiH/8-refactoring-previous-ticket-to-polymorphism

Alexis220382 commented on src/com/peterson/util/Util.java:
```diff
@@ -16,13 +16,21 @@
+        System.out.println(String.format("Deleting element %d from Collection takes: %d", i, (stop-start)));
+    }
+
+    public static void deleteObject(HashSet<Integer> collection, Object o){
```
Переменная `collection` не отражает сущности переменной, назови ее по-другому.
См. `class WorlWithCollection`

ap2144839 committed:
refactoring-previous-ticket-to-polymorphism
- class com.peterson.collection.CollectionFactory was created;
- enum com.peterson.collection.CollectionType was created;
- class com.peterson.collection.WorkWithCollection was modified as abstract ;
- class com.peterson.collection.WorkWithList was created to extend abstract class com.peterson.collection.WorkWithCollection;
- class com.peterson.collection.WorkWithHashSet was created to extend abstract class com.peterson.collection.WorkWithCollection;
- class com.peterson.util.Util was changed in order to use the methods from above classes
https://trello.com/c/w8bRuNiH/8-refactoring-previous-ticket-to-polymorphism

Alexis220382 reviewed (approved):
Че-то ты опять намутил с новым параметром, но ладно, все позже станет понятно.
Мержи в мастер
